server.js

```diff
@@ -2,29 +2,80 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import harryData from './data/harry-potter-characters.json'
+
+const ERROR_CHARACTERS_NOT_FOUND = {error : 'No character results were found, please try again.'}
+
+// Defines the port the app will run on
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// Middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// Starting route
+app.get('/', (request, response) => {
+  response.send('Hello, this is an API with Harry Potter characters. Go to https://lindas-project-express-api.herokuapp.com/characters to see all the characters. For more documentation, visit: https://github.com/lindahz/project-express-api')
+})
+
+// This endpoint shows all characters + will let you filter on name with query parameter
+app.get('/characters', (request, response) => {
+  const { name } = request.query
+  const filteredName = harryData.filter(character => 
+    character.name.toLocaleLowerCase() && 
+    character.name.toLocaleLowerCase().includes(name))
+
+  if (filteredName.length === 0) {
+    response.json(harryData)
```

Comment:
We shouldn't return list of all results if we didn't fulfill requirements of query.
If there were no results for our query (so, when `filteredName.length === 0`), we should return empty array. This is very important info for the user - "Aha, so I queried for character "Parry Hotter" and got empty array. That clearly means there is no book in database that has "Parry Hotter" character"

Reply:
*** Needs to be fixed ***